internal/armadactl/analyze.go

```diff
@@ -6,30 +6,26 @@ import (
 	"fmt"
 	"reflect"
 
-	"google.golang.org/grpc"
-
 	"github.com/G-Research/armada/pkg/api"
 	"github.com/G-Research/armada/pkg/client"
 	"github.com/G-Research/armada/pkg/client/domain"
 )
 
 func (a *App) Analyze(queue string, jobSetId string) error {
 	fmt.Fprintf(a.Out, "Querying queue %s for job set %s\n", queue, jobSetId)
-	client.WithConnection(a.Params.ApiConnectionDetails, func(conn *grpc.ClientConn) {
-		eventsClient := api.NewEventClient(conn)
-
+	return client.WithEventClient(a.Params.ApiConnectionDetails, func(c api.EventClient) error {
 		events := map[string][]*api.Event{}
 		var jobState *domain.WatchContext
 
-		client.WatchJobSet(eventsClient, queue, jobSetId, false, true, context.Background(), func(state *domain.WatchContext, e api.Event) bool {
+		client.WatchJobSet(c, queue, jobSetId, false, true, context.Background(), func(state *domain.WatchContext, e api.Event) bool {
```

Comment:
Can we use a more descriptive name than c?

Reply:
How about `ec`? The scope of the variable is quite limited.